onnx/test/serialization_test.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) ONNX Project Contributors
+
+# SPDX-License-Identifier: Apache-2.0
+import os
+import tempfile
+import unittest
+
+import onnx
+
+_TEST_MODEL = """\
+<
+    ir_version: 8,
+    opset_import: ["" : 17, "local" : 1]
+>
+agraph (float[N] X) => (float[N] Y) {
+    Y = local.foo (X)
+}
+
+<opset_import: ["" : 17, "local" : 1], domain: "local">
+foo (x) => (y) {
+    temp = Add(x, x)
+    y = local.bar(temp)
+}
+
+<opset_import: ["" : 17], domain: "local">
+bar (x) => (y) {
+    y = Mul (x, x)
+}"""
+
+
+class _OnnxTestTextualSerializer(onnx.serialization.ProtoSerializer):
+    """Serialize and deserialize the ONNX textual representation."""
+
+    supported_formats = ("onnxtext",)
+
+    def serialize_proto(self, proto) -> bytes:
+        text = onnx.printer.to_text(proto)
+        return text.encode("utf-8")
+
+    def deserialize_proto(self, serialized: bytes, proto):
+        text = serialized.decode("utf-8")
+        if isinstance(proto, onnx.ModelProto):
+            return onnx.parser.parse_model(text)
+        if isinstance(proto, onnx.GraphProto):
+            return onnx.parser.parse_graph(text)
+        if isinstance(proto, onnx.FunctionProto):
+            return onnx.parser.parse_function(text)
+        if isinstance(proto, onnx.NodeProto):
+            return onnx.parser.parse_node(text)
+        raise ValueError(f"Unsupported proto type: {type(proto)}")
+
+
+class TestRegistry(unittest.TestCase):
+    def setUp(self) -> None:
+        self.serializer = _OnnxTestTextualSerializer()
+        onnx.serialization.registry.register(self.serializer)
+
+    def test_get_returns_the_registered_instance(self) -> None:
+        serializer = onnx.serialization.registry.get("onnxtext")
+        self.assertIs(serializer, self.serializer)
+
+    def test_get_raises_for_unsupported_format(self) -> None:
+        with self.assertRaises(ValueError):
+            onnx.serialization.registry.get("unsupported")
+
+    def test_onnx_save_load_model_uses_the_custom_serializer(self) -> None:
```

Comment:
Is there a way we can use the serializer without saving on disk?

Reply:
Added tests